### DISCUSSION TITLE
Discuss Enhancement Request

### DISCUSSION THREAD

**Discussion by stuartlewis06** [spotbugs/spotbugs#3234] [Ideas]
I'd like to merge an enhancement into the spotbugs repo. The suggested enhancement enables logging the spotbugs violation count out to plain file. This can then be used as part of a CI pipeline. (The proposed changes have been added to a forked repository which we have been successfully utilizing for almost a year now).

In terms of next steps, I'd like to raise an enhancement request but this message serves to hopefully enable discussion on the proposed enhancement prior to potentially moving forwards with this. Could I kindly ask for your thoughts on this?

Kind Regards,
Stuart Lewis

**Comment by gtoison** [comment]
Hello,
Spotbugs currently exports its findings in three standard formats (HTML, Sarif and XML), and also a few custom: Emacs, XDocs
Are you proposing to add another export, presumably extending `BugCollectionBugReporter`?

My intuition is that is it's might be better to support a user-provided exporter, because we might lose track of your requirements. Formatting data is not so easy, especially when there isn't a specification you can refer to.

Or maybe
